packages/shared/lib/core/ledger/utils/openLedgerConfirmationPopup.ts

```diff
@@ -0,0 +1,37 @@
+import { PopupProps } from '@auxiliary/popup'
+import { isPreparedTransaction, isPreparedTransactionEssenceHash } from '@core/profile-manager'
+import { TransactionProgressEventPayload } from '@core/profile-manager/api/types'
+import { openPopup } from '@lib/popup'
+
+import { deconstructLedgerSendConfirmationProps } from '../helpers'
+
+export function openLedgerConfirmationPopup(
+    payload?: TransactionProgressEventPayload,
```

Comment:
It's for a guard to check that the payload is really there. We've shot ourselves in the foot many times passing `undefined` values to functions where it expected something and because we didn't check that the object was really there Firefly would crash.

Reply:
But this doesn't solve the issue, now it just fails silently. Maybe we should throw an error if there is no payload?

Reply:
Yeah that wouldn't be a bad idea, I'll add really quick